vars/skipStage.groovy

```diff
@@ -158,16 +156,31 @@ boolean skip_ftest_valgrind(String distro, String target_branch) {
 }
 
 boolean skip_ftest_hw(String size, String target_branch) {
-    return already_passed() ||
+    // Skip the functional hardware test stage if it has already passed or
+    // there are no tests matching the tags for the stage
+    if (already_passed() || !testsInStage()) {
+        return true
+    }
+    // Run the functional hardware test stage if explicitly requested by the user
+    if (run_default_skipped_stage('func-hw-test-' + size) ||
+        run_default_skipped_stage('func-test-hw-' + size) ||
+        cachedCommitPragma('Run-daily-stages').toLowerCase() == 'true') {
+        // Forced to run due to a (Skip) pragma set to false
+        return false
+    }
+    String distro = hwDistroTarget()
```

Comment:
Since you already have `size` in this function, it would be more optimal to pass it to `hwDistroTarget()` but more importantly, you should be reusing this `distro` variable down on line 185 where `hwDistroTarget(size)` is being called again.

Reply:
Ah right, thanks.